### DISCUSSION TITLE
Minimum Ratio?

### DISCUSSION THREAD

**Discussion by Friday13th87** [StuffAnThings/qbit_manage#449] [Q&A]
Is there a way to set the minimum ratio a torrent needs to have before some action is done?
Something like minimum ratio and minimum seedtime, if both are reached (not only one of both) the torrent is deleted, paused, etc?

**Comment by bakerboy448** [comment]
max seeding ratio in combination with min seeding time for share limits will create an AND condition

**Comment by Friday13th87** [comment]
but this AND condition is not comparable to a min seeding ratio.

case 1:
condition: 600 minutes min seeding time and 2 max ratio
--> after 600 minutes with a ratio of 0.5 both conditions are met and the torrent would be deleted for example. 0.5 is less then the max ratio of 2.

case 2:
same condition: 600 minutes min seeding time and 2 max ratio
--> after 600 minutes with a ratio of 5, the 600 minutes condition is met, but the ratio of 5 is higher then the max ratio. no glue if its met then or not.


not what i was searching for sadly.

**Reply by bakerboy448** [reply]
> after 600 minutes with a ratio of 0.5 both conditions are met and the torrent would be deleted for example. 0.5 is less then the max ratio of 2.

Incorrect. No idea where you got this from? Please read the config file and wiki that explains these attributes

Seed for at least 600 min and a ratio of 2. Exactly what you want

> after 600 minutes with a ratio of 5, the 600 minutes condition is met, but the ratio of 5 is higher then the max ratio. no glue if its met then or not.

seeded for at least 600min (min seed time) AND seed to at least 2 (max seed ratio), exactly as you want

**Comment by Friday13th87** [comment]
Nope, sorry, its absolutley not clear.
the wiki says:
`Will set the torrent Maximum share ratio until torrent is stopped from seeding/uploading.`

the config example says:

```
   # <OPTIONAL> max_ratio <float>: Will set the torrent Maximum share ratio until torrent is stopped from seeding/uploading.
    # Will default to -1 (no limit) if not specified for the group.
    max_ratio: 5.0
    # <OPTIONAL> max_seeding_time <int>: Will set the torrent Maximum seeding time (minutes) until torrent is stopped from seeding.
    # Will default to -1 (no limit) if not specified for the group.
    max_seeding_time: 129600
    # <OPTIONAL> min_seeding_time <int>: Will prevent torrent deletion by cleanup variable if torrent has not yet minimum seeding time (minutes).
```
The min seeding time will prevent torrent deletion if this value is not met.
This implements for me that the max ratio is only preventing of further seeding but not preventing a torrent if the min seed time is reached to get cleaned up. having a min and max seed time but only a max ratio is saying that the min ratio is missing.
if the "max ratio" is implementing a "min ratio" as well, then it makes no sense to call it even a "max ratio", then it would only be the "ratio"

but if it is really true what you are saying, then it will work how i wished it would be possible.

but from reading the wiki i dont actually think that you are right, otherwise someone should rewrite it.

**Reply by bakerboy448** [reply]
you also missed cleanup

> Setting this as true will remove any torrents that satisfies the share limits. It will also delete the torrent's data if and only if no other torrents are using the same folder/files.


Max ratio and max seed time are the seed goals / share limits which are OR just like qbit and Starr....they are the max ratio/seed time you want a torrent to reach before it's removed.

min seed time means the torrent must seed for that minimum time agnostic of any seed goals

wiki is open source, PRs welcome....same for the config file.

**Reply by albelem** [reply]
I realize this is a bit of an old topic, but I just got precisely the behavior @Friday13th87 was referring to:
I have an entry with the following options:
```
     max_seeding_time: 44000
     min_seeding_time: 10000
     min_num_seeds: 2
     max_ratio: 1.0
     cleanup: true
 ```
 What I'm trying to achieve:

- if the torrent has been seeding for 30 days AND has a minimum of 2 seeds AND has at least 1.0 ratio: stop it and delete
- the torrent needs to seed at least 6 days, if those 6 days are past and it has a min. 2 seeds and 1.0 ratio: stop it and delete
- if the torrent doesn't have have at least 2 seeds AND 1.0 ratio never stop it or delete it

What I just witness happening: torrent started 1 hour ago, reached 1.0 ratio and I found it stopped on the client.

**Reply by bakerboy448** [reply]
> if the torrent has been seeding for 30 days AND has a minimum of 2 seeds AND has at least 1.0 ratio: stop it and delete

if you want to seed for at least 30 days then you must set 30 days as your minimum seed time.

Max Ratio and Max Seed Time are ALWAYS ORs

> the torrent needs to seed at least 6 days, if those 6 days are past and it has a min. 2 seeds and 1.0 ratio: stop it and delete

min seed time 6 days; min seeders 2; max ratio 1

> if the torrent doesn't have have at least 2 seeds AND 1.0 ratio never stop it or delete it

min seeders 2 and max ratio 2.0.
It's impossible to never stop it, but QBm will remove the limits on next run

> torrent started 1 hour ago, reached 1.0 ratio and I found it stopped on the client.

this is expected and how it always works. It will not change as this is how qbit operates, take it up with them if it's an issue.   On next run QBm will remove the limits and resume the torrent if you've enabled it to do so for the share limit

**Reply by albelem** [reply]
ohhhhhhhhhhhh so on next run it'll get resumed ? that is interesting. So maybe I should be running more often so that it spends less time paused.
Thanks for the clarification.

**Reply by bakerboy448** [reply]
Yes. I run every 90 min
